qiskit_optimization/algorithms/optimization_algorithm.py

```diff
@@ -376,7 +378,10 @@ def _prepare_converters(converters: Optional[Union[QuadraticProgramConverter,
             return [converters]
         elif isinstance(converters, list) and \
                 all(isinstance(converter, QuadraticProgramConverter) for converter in converters):
-            return converters
+            if any(isinstance(converter, MaximizeToMinimize) for converter in converters):
+                return converters
+            else:
+                return converters + [MaximizeToMinimize()]
```

Comment:
Do we need a `MaximizeToMinimize` converter in every optimizer by default?
@t-imamichi what do you think?

Reply:
I don't think we need to append max2min here. But, we need to raise an error if an algorithm without max2min receives maximization problem.

Reply:
Okay @t-imamichi , I will remove the appending of Max2Min to the list. 
I would propose error raising in the following optimizers:
grover, minimum_eigen_optimizer

We could do this after conversion of the problem in ``solve`` method , e.g. after
https://github.com/Qiskit/qiskit-optimization/blob/724158ba42f4b75758edc1de42f6f665660ec27e/qiskit_optimization/algorithms/grover_optimizer.py#L175-L176

        # convert problem to minimization QUBO problem
        problem_ = self._convert(problem, self._converters)
        if problem_.objective.sense == QuadraticObjective.Sense.MAXIMIZE:
            raise QiskitOptimizationError(
                "Optimizer internally deals with minimization problem only. \
                Please ensure conversion."
            )

What do you think?

There are 2 more optimizers without max2min: recursive_meo, warm_start_qaoa

Recursive meo uses other optimizers and should therefore not throw an error. The error would be thrown eventually by those optimizers solve methods.

Warm_start_qaoa uses an optimizer as pre-solver. Like described in my comment(shortly after PR) this can create conflicts, for example with ``GoemansWilliamsonOptimizer``. We should let the user take care by an appropriate converters list.

Reply:
The compatibility is checked by `OptimizationAlgorithm._verify_compatibility` and internally `get_compatibility_msg` returns some message if the problem is not compatible. For GroverOptimizer, it checks the compatibility as follows.
https://github.com/Qiskit/qiskit-optimization/blob/6be880b992d9a2f1de8cc89cfd999b786f98ce8a/qiskit_optimization/algorithms/grover_optimizer.py#L107-L119

If the default converter is used in GroverOptimizer, we don't need to take care of the error handling. But, if a customized converter is given, it gets tricky. Let me think.

Reply:
We have to think of the case where a customized converter is given to `GroverOptimizer` in a separate PR because the current PR is already large enough.

I recommend the following to keep this PR simple.
1. not add an extra check of compatibility than `_verify_compatibility` in this PR
2. revert `_prepare_converters` not to append max2min.

Reply:
Converter max2min is not appended to the list anymore and there is no extra check.